pkg/metrics/github_fetcher.go

```diff
@@ -12,10 +12,35 @@ import (
 )
 
 var (
-	repositories []string
-	workflows    map[string]map[int64]github.Workflow
+	repositories  []string
+	repos_per_org map[string]int
+	workflows     map[string]map[int64]github.Workflow
 )
 
+func countAllReposForOrg(orga string) int {
+	query := "org:" + orga
+	opt := &github.SearchOptions{
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+			Page:    1,
+		},
+	}
+
+	for {
+		result_page, _, err := client.Search.Repositories(context.Background(), query, opt)
```

Comment:
Search's ratelimit is distinct from the overall one, and is more locked down at 10 requests/min ([source](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#rate-limit)). the low time between refreshes will probably help here, but i think there's a better way to get just the repo count:
[GET Organization](https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#get-an-organization)

Reply:
this returns `total_private_repos` and `owned_private_repos`, and i think you want to add them both (`total_private_repos` == number of repos w/ the "Private" visibility; `total_private_repos` includes "Internal")

tho i'm a little confused about the exact number, b/c when i run it i get the right number of private repos from the API vs the UI, `43`, but i get `1389` for `total_private_repos`, but the UI reports `1,349` which means that the `total_private_repos` is reporting 3 less b/c there is no overlap in the UI, but added together makes `1392`

Reply:
looking at the return of
```sh
gh api -X GET search/repositories -f q='org:wbd-streaming'
```
i see it returns 1389, so `total_private_repos` is equivalent

Reply:
The search endpoint is limited to `30 requests per minute` for non-code-related searches, I'd need to check if the Repo search looks internally for the Code search but your point on migrating to the Org is valid.

Let me run some tests to see the payload and do the math